File: inject_sidebar.py
import os
import glob

base_path = '/Users/ojaeeul/Downloads/세종요리제과학원/무제 폴더/수정전/sejk 4/sejongcook-next/Sejong/SejongAttendance/public'
html_files = glob.glob(os.path.join(base_path, '*.html'))

inject_str = """
                    <a href="attendance_manager.html" id="navAttendanceManager" class="nav-item">
                        <span class="material-icons" style="font-size: 1.2rem; margin-right: 8px;">person_remove</span>
                        학생별 출석 관리
                    </a>"""

target_str = '<a href="index.html?filter=archive" id="navArchive" class="nav-item">수료생 보관함</a>'

for file_path in html_files:
    with open(file_path, 'r', encoding='utf-8') as f:
        content = f.read()
    
    if target_str in content and 'id="navAttendanceManager"' not in content:
        content = content.replace(target_str, target_str + inject_str)
        with open(file_path, 'w', encoding='utf-8') as f:
            f.write(content)
        print(f"Injected into {os.path.basename(file_path)}")

File: inject_sidebar.js
const fs = require('fs');
const path = require('path');

const dir = '/Users/ojaeeul/Downloads/세종요리제과학원/무제 폴더/수정전/sejk 4/sejongcook-next/Sejong/SejongAttendance/public';
const files = fs.readdirSync(dir).filter(f => f.endsWith('.html'));

const injectTag = '<script src="sidebar_sync.js?v=20260622"></script>';

let modifiedCount = 0;
files.forEach(file => {
    const filePath = path.join(dir, file);
    let content = fs.readFileSync(filePath, 'utf8');
    
    if (content.includes('sidebar_sync.js')) {
        console.log(`Skipping ${file} (already injected)`);
        return;
    }
    
    // Inject right before </body>
    if (content.includes('</body>')) {
        content = content.replace('</body>', `    ${injectTag}\n</body>`);
        fs.writeFileSync(filePath, content, 'utf8');
        console.log(`Injected into ${file}`);
        modifiedCount++;
    } else {
        console.log(`Could not find </body> in ${file}`);
    }
});

console.log(`Successfully injected into ${modifiedCount} files.`);
